nexus/db-queries/src/db/datastore/switch_port.rs

```diff
@@ -25,16 +28,72 @@ use diesel::{
     CombineDsl, ExpressionMethods, JoinOnDsl, NullableExpressionMethods,
     QueryDsl, SelectableHelper,
 };
+use ipnetwork::IpNetwork;
+use nexus_db_model::{
+    SqlU16, SqlU32, SqlU8, SwitchPortBgpPeerConfigAllowExport,
+    SwitchPortBgpPeerConfigAllowImport, SwitchPortBgpPeerConfigCommunity,
+};
 use nexus_types::external_api::params;
 use omicron_common::api::external::http_pagination::PaginatedBy;
 use omicron_common::api::external::{
-    self, CreateResult, DataPageParams, DeleteResult, Error, ListResultVec,
-    LookupResult, NameOrId, ResourceType, UpdateResult,
+    self, CreateResult, DataPageParams, DeleteResult, Error,
+    ImportExportPolicy, ListResultVec, LookupResult, NameOrId, ResourceType,
+    UpdateResult,
 };
 use ref_cast::RefCast;
 use serde::{Deserialize, Serialize};
 use uuid::Uuid;
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct BgpPeerConfig {
```

Comment:
This data structure is mostly the same as the underlying db-model data structure, but includes list data members (`allowed_import`, `allowed_export`, `communities`) which are not in the db-model structure as they've been normalized out into separate tables and thus have their own data structures. This makes it so we can provide a complete view of the underlying BGP peer config to calling clients through this type.